File: kotlinames-common/src/main/java/com/nagopy/android/kotlinames/property/required/KRequiredBooleanPropertyName.java
package com.nagopy.android.kotlinames.property.required;

import com.nagopy.android.kotlinames.KRequiredPropertyName;
import com.nagopy.android.kotlinames.KSortablePropertyName;

public class KRequiredBooleanPropertyName implements KRequiredPropertyName<Boolean>, KSortablePropertyName<Boolean> {

    private final String name;

    public KRequiredBooleanPropertyName(String name) {
        this.name = name;
    }

    @Override
    public String name() {
        return name;
    }
}

File: kotlinames/src/main/kotlin/com/nagopy/android/kotlinames/property/required/KRequiredBooleanPropertyName.kt
package com.nagopy.android.kotlinames.property.required

import com.nagopy.android.kotlinames.KRequiredPropertyName
import com.nagopy.android.kotlinames.KSortablePropertyName

class KRequiredBooleanPropertyName(val name: String) : KRequiredPropertyName<Boolean>, KSortablePropertyName<Boolean> {

    override fun name(): String = name

}
